### PR TITLE
Add test case for https://github.com/bazelbuild/bazel/issues/15053

### PR DESCRIPTION
**Pull request by guw**
Test case for reproducing the parser problem reported in https://github.com/bazelbuild/bazel/issues/15053

### CODE REVIEW

**Comment by guw** [comment]
Debugging the test case unveils:  it's the `[]` array indicator that's not handled well by the parser. In the context of the annotation, the value is actually the type `SomeClass[].class`. I'll simplify the test data.

**Comment by guw** [comment]
@cushon Apologies for the direct ping. Would you be able to find someone helping me fix this?

File: java/com/google/turbine/parse/Token.java [hunk 1]
```diff
@@ -23,8 +23,8 @@
   RPAREN(")"),
   LBRACE("{"),
   RBRACE("}"),
-  LBRACK("<"),
-  RBRACK(">"),
+  LBRACK("["),
+  RBRACK("]"),
```
Comment (guw) [inline java/com/google/turbine/parse/Token.java:27]: @cushon I wasn't sure about this change. Do you know why the value was off/wrong?

File: java/com/google/turbine/parse/ConstExpressionParser.java [hunk 2]
```diff
@@ -455,6 +452,9 @@ private long parseLong(String text, int radix) {
       }
       eat();
     }
+    if (token == Token.LBRACK) {
+        return finishClassLiteral(pos, asClassTy(pos, bits.build()));
+    }
```
Comment (guw) [inline java/com/google/turbine/parse/ConstExpressionParser.java:457]: @cushon Moving this block below the `while` loop fixes the test. All other tests still pass.

Comment (cushon) [reply java/com/google/turbine/parse/ConstExpressionParser.java:457]: Thanks for investigating this and for the report. I started looking at in parallel and came to a similar conclusion.

File: java/com/google/turbine/parse/Token.java (hunk 1 above)
Comment (cushon) [reply java/com/google/turbine/parse/Token.java:27]: Huh, nice catch. I think these strings are just mostly used for debugging, and this was an oversight that went unnoticed

**Pull request closed without merging**